modules/actions/pool/sync-swaps.ts

```diff
@@ -0,0 +1,74 @@
+import { Chain, Prisma } from '@prisma/client';
+import { prisma } from '../../../prisma/prisma-client';
+import { V3VaultSubgraphClient } from '../../sources/subgraphs';
+import _ from 'lodash';
+import { swapTransformer } from '../../sources/transformers/swap-transformer';
+import { OrderDirection, Swap_OrderBy } from '../../sources/subgraphs/balancer-v3-vault/generated/types';
+import { swapsUsd } from '../../sources/enrichers/swaps-usd';
+
+/**
+ * Adds all swaps since daysToSync to the database. Checks for latest synced swap to avoid duplicate work.
+ *
+ * @param vaultSubgraphClient
+ * @param chain
+ * @param daysToSync
+ * @returns
+ */
+export async function syncSwaps(
+    vaultSubgraphClient: V3VaultSubgraphClient,
+    chain = 'SEPOLIA' as Chain,
+    daysToSync = 30,
+): Promise<string[]> {
+    const vaultVersion = 3;
+
+    // Get latest event from the DB
+    const latestEvent = await prisma.poolEvent.findFirst({
+        where: {
+            type: 'SWAP',
+            chain: chain,
+            vaultVersion,
+        },
+        orderBy: {
+            blockNumber: 'desc',
+        },
+    });
+
+    // Get events since the latest event or limit to number or days we want to keep them in the DB
+    const since = Math.floor(+new Date(Date.now() - daysToSync * 24 * 60 * 60 * 1000) / 1000);
+    const where =
+        latestEvent?.blockTimestamp && latestEvent?.blockTimestamp > since
+            ? { blockNumber_gte: String(latestEvent.blockNumber) }
+            : { blockTimestamp_gte: String(since) };
+
+    // Get events
+    const { swaps } = await vaultSubgraphClient.Swaps({
+        first: 1000,
+        where,
+        orderBy: Swap_OrderBy.BlockNumber,
+        orderDirection: OrderDirection.Asc,
+    });
+
+    // Store only the events that are not already in the DB
+    const existingEvents = await prisma.poolEvent.findMany({
+        where: {
+            id: { in: swaps.map((event) => event.id) },
```

Comment:
why do we need this? we already only get swaps with a greater blockNumber than what we have stored.

Reply:
maybe we dont even need to check against existing events? just upsert or create with skipDuplicates?

Reply:
does prisma detects duplicates by ID? if so, probably not needed.

Reply:
although it's useful, because we don't need to be getting USD prices for synced events.

Reply:
I'd remove it completely, there might be some duplicate swaps but it would save another potentionally big query (as id: { in: XXX} is a huge array)